src/app/auth.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { Http, Headers, Response } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+
+import { AppConfig } from './app.config'
+
+@Injectable()
+export class AuthService {
+
+  constructor(private http: Http, private config: AppConfig) { }
+
+  login(username: string, password: string, remember: boolean) {
+    var headers = new Headers();
+    headers.append('Content-Type', 'application/json');
+    return this.http.post(this.config.apiUrl + '/account/token', JSON.stringify({ username: username, password: password }), {headers: headers})
+      .map((response: Response) => {
+        let authtoken = response.json();
+        if (authtoken) {
```

Comment:
The server response is in the form of 
`{ authToken: ... }`
so this will be true even if a completely different object is returned or if the actual authToken is null or an empty string. Maybe this should rather be something like:
```
let res = response.json();
if (res.authToken)
```
I know it's an unlikely case but it might cause a hard to find bug in future.